### PR TITLE
Proposal for implementing HA for ruler

### PR DESCRIPTION
**Pull request by soonping-amzn**
**What this PR does**:
Adds a proposal to for implementing HA in ruler

**Which issue(s) this PR fixes**:
Consolidates multiple issues and PRs:
- [Issue #4435](https://github.com/cortexproject/cortex/issues/4435)
- [PR #4712](https://github.com/cortexproject/cortex/pull/4712)
- [PR #4768](https://github.com/cortexproject/cortex/pull/4768)

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by alvinlin123** [comment]
This proposal looks ok to me, so I am ok to merge this in as "proposed". Further discussion with the community are needed to switch the status to "accepted".

**Review by alvinlin123 — APPROVED** [review]

File: docs/proposals/ruler-ha.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+title: "Ruler HA"
+linkTitle: "Ruler HA"
+weight: 1
+slug: ruler-ha
+---
+
+- Author: [Soon-Ping Phang](https://github.com/soonping-amzn)
+- Date: Jube 2022
```
Comment (roystchiang) [inline docs/proposals/ruler-ha.md:9]: June

File: docs/proposals/ruler-ha.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+This proposal consolidates multiple existing PRs from the AWS team working on this feature, as well as future work needed to complete support.  The hope is that a more holistic view will make for more productive discussion and review of the individual changes, as well as provide better tracking of overall progress.
+
+The original issue is [#4435](https://github.com/cortexproject/cortex/issues/4435).
+
+## Problem
+
+Rulers in Cortex currently run with a replication factor of 1, wherein each RuleGroup is assigned to exactly 1 ruler.  This lack of redundancy creates the following risks:
+
+- Rule group evaluation
+  - Missed evaluations due to a ruler outage, possibly caused by a deployment, noisy neighbour, hardware failure, etc.
```
Comment (roystchiang) [inline docs/proposals/ruler-ha.md:24]: could you give a brief description of how this proposal mitigates this issue?

File: docs/proposals/ruler-ha.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  - Missed evaluations due to a ruler brownout due to other tenant rule groups sharing the same ruler (noisy neighbour)
```
Comment (roystchiang) [inline docs/proposals/ruler-ha.md:25]: maybe not within the scope of this proposal but is it possible to detect the impact some how? 

for example, when all the rules for a single tenant is impacted, do we have a way to find out what the impact was?

File: docs/proposals/ruler-ha.md (hunk 2 above)
Comment (soonping-amzn) [reply docs/proposals/ruler-ha.md:24]: I've added some text in the ReplicationFactor parameter section to address this.

File: docs/proposals/ruler-ha.md (hunk 1 above)
Comment (soonping-amzn) [reply docs/proposals/ruler-ha.md]: Fixed

**Review by friedrichg — APPROVED** [review]
LGTM

**Pull request merged**